### DISCUSSION TITLE
Initial release :tada:

### DISCUSSION THREAD

**Discussion by diazona** [diazona/setuptools-pyproject-migration#92] [General]
I'm making this as a convenient place for any miscellaneous discussions that come up during the process of creating an initial release. It may wind up being totally unnecessary, but if we do need to coordinate or discuss anything about the procedure we can do it here. FYI @sjlongland

**Comment by diazona** [comment]
I'm thinking we can actually make an alpha test release without the changelog being ready. It'll be just a test run of the real initial release which will come out once we've got the changelog finalized, so the alpha release won't cause us to close the milestone. If you don't mind @sjlongland how about I do that today? Then if any problems show up in the release procedure we can fix them before publishing the real initial release.

**Reply by diazona** [reply]
actually I had not noticed you already pushed the changelog updates to #93 before I posted this... my bad, I think I somehow missed a chunk of notifications.

Anyway I'll go ahead and make that alpha release to see if anything breaks, but let's wait to make a full release for a day or two just so we have a chance to do quick sanity checks.

**Reply by sjlongland** [reply]
No biggie, the changes to the `CHANGELOG.rst` should not make any difference to the code base.  Looks like we've got some fun and games to troubleshoot regarding pushing to pypi anyway.

I've given it a poke with debugging enabled.

**Comment by diazona** [comment]
Time to make the final release? Or should we do another trial run? (`v0.1a1` perhaps... though it occurs to me that these are more properly "release candidates" so I guess we should have used `v0.1rc0` and `, but it's not a big deal)

**Reply by diazona** [reply]
...oh wait, I forgot, we have to aggregate the changelog first. Let me push a PR for that.

**Reply by diazona** [reply]
Done in #97, and once that's merged, feel free to take a run at making a release.

I would suggest copying the release notes from [the prerelease](https://github.com/diazona/setuptools-pyproject-migration/releases/tag/v0.1a0) and just adding the one latest entry for this last PR, because (as explained in the PR and commit message) I think it'll be most useful to have the release notes list diffs from one final release to the next, not counting prereleases.

**Reply by sjlongland** [reply]
Yep, probably worth me giving it a shot… I've got a review to do then I should get onto this.

**Reply by diazona** [reply]
How's the progress on that? (Not that we're in a hurry, really, I'm just excited to get the final release out and start spreading the word :smile:)

**Reply by diazona** [reply]
...it just occurred to me how weird it is that this is both the "initial release" and a "final release" :laughing:

**Reply by sjlongland** [reply]
Ahh right… yes… got side-tracked.  As for it being the final release, there's always bug fixes. :-)

**Reply by sjlongland** [reply]
Well, this worked well… :-)

![image](https://github.com/diazona/setuptools-pyproject-migration/assets/1710526/03335e8d-71f1-4c75-b871-b1645c825289)

Does one create and push the tag first, _then_ create a release from it?

**Reply by sjlongland** [reply]
Nope, can't do it that way either.

```
RC=0 stuartl@rikishi ~/projects/setuptools-pyproject-migration/work $ git push --tags origin
Enumerating objects: 1, done.
Counting objects: 100% (1/1), done.
Writing objects: 100% (1/1), 167 bytes | 167.00 KiB/s, done.
Total 1 (delta 0), reused 0 (delta 0), pack-reused 0
remote: error: GH006: Protected tag update failed for refs/tags/v0.1.0.
remote: error: You're not authorized to create a tag
To github.com:diazona/setuptools-pyproject-migration.git
 ! [remote rejected] v0.1.0 -> v0.1.0 (protected tag hook declined)
error: failed to push some refs to 'github.com:diazona/setuptools-pyproject-migration.git'
```

**Reply by diazona** [reply]
Ah, darn, I forgot about that: it looks like you only have collaborator access to the repo, whereas we have a tag protection rule that requires maintainer access or some kind of special permission to create tags. I'll try to figure out how to make you a maintainer, but I seem to remember trying to do that before and it being difficult, so for now I'll just delete the tag protection rule. That will allow any collaborator to create tags (and releases), but it's fine as long as it's just the two of us who are collaborators.

Give it another try now that the protection rule is deleted.

**Reply by diazona** [reply]
After a quick documentation search, it looks like I [actually can't add maintainers to a personal project](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-user-account-settings/permission-levels-for-a-personal-account-repository#about-permissions-levels-for-a-personal-account-repository).

I _do_ actually have [an organization](https://github.com/ellipsix/) which I never really meant to use for much, but I could transfer the repo to that and that would give us more flexibility to hand out permissions. What do you think? (I believe this can wait until after the release - we'll need to update some URLs and stuff to account for the switch but it shouldn't be too much of a problem otherwise.)

**Reply by sjlongland** [reply]
Yep, well that could be an option, we transfer it to an organisation… whether it be the one you mentioned, or whether we create a dedicated one for the project.

But, first… let's see if we can get v0.1.0 out the door. :-)

**Reply by sjlongland** [reply]
Ahh, this looks a lot better…
![image](https://github.com/diazona/setuptools-pyproject-migration/assets/1710526/9692d267-fc89-48d0-a0f0-52e93abcf18f)

**Reply by diazona** [reply]
Hooray!

**Comment by diazona** [comment]
Now that the release is out, shall we share the news on Mastodon? It seems fitting to start there since that's where the idea for this project hatched :smile: 

At some point I think it'd make sense to email the python-announce list as well, but I was thinking we could get some early feedback from a smaller audience first and address any glaring issues.

**Reply by sjlongland** [reply]
I think so, get a wider audience out there.

**Reply by diazona** [reply]
You know what, maybe I'll wait until tomorrow my time to post on Mastodon because North American users will be more active then. But you can post something now to catch the Aus/NZ/East Asian crowd, or later, or whatever you like (there's no real _need_ for us to coordinate).
